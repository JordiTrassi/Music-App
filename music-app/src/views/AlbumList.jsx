import * as React from 'react';
import { Box, Button, CircularProgress, List, ListItem, ListItemAvatar, ListItemText, styled, Typography } from '@mui/material';
import { useDispatch, useSelector } from 'react-redux';
import { getAlbums } from '../store';
import { AlbumCard } from '../components';

const Img = styled('img')({
  margin: 'auto',
  display: 'block',
  maxWidth: '100%',
  maxHeight: '100%',
});



export const AlbumList = () => {

    const dispatch = useDispatch();
    const {isLoading, albums = [], page } = useSelector(state => state.playList);

    return (
      <Box>
        {/* <CircularProgress disableShrink={!isLoading} />   */}
            <List sx={{ width: '100%', maxWidth: '80%', bgcolor: 'background.paper', pl: 5 }}>
                
                {
                    albums.map( album => (
                        
                        <AlbumCard                    
                            key={album.trackId}
                            {...album}
                        />
                    ))
                        
                }
            </List>
            
            <Button
                variant="contained"
                disabled={isLoading}
                onClick={() => dispatch(getAlbums(page))}
                sx={{ p: 2, marginLeft: '50%', marginBottom: '30px' }}
            >
                Next Page
            </Button>
      </Box>
  );
}


