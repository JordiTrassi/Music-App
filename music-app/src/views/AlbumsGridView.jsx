

export const AlbumsGridView = () => {
  return (
    <div>
      
    </div>
  )
}

