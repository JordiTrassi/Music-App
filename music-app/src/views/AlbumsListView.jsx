import { useSelector } from 'react-redux';
import { List} from '@mui/material';
import { AlbumCard, IsLoading } from '../components';


export const AlbumsListView = () => {
    
    const {isLoading, albums = [], page, viewGrid  } = useSelector(state => state.playList);


    return (
        
        <List sx={{ width: '100%', bgcolor: 'background.paper', p: 1 }}>
            
            {
                (isLoading)
                    ? <IsLoading />
                    : albums.map(album => (
                        <AlbumCard                    
                            key={album.collectionId}
                            {...album}
                        />
                        ))
                    
            }
        
        </List>
    );
}


