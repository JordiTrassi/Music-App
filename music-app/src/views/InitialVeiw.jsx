import { MusicNote } from "@mui/icons-material"
import { Grid, Typography } from "@mui/material"


export const InitialVeiw = () => {
  return (
    <Grid
        className='animate__animated animate__fadeIn animate__faster'
        container
        spacing={0}
        direction="column"
        alignItems="center"
        justifyContent="center"
        sx={{ minHeight: 'calc(100vh - 110px)', backgroundColor: 'primary.main', borderRadius: 3}}
      >
        <Grid item xs={12}>
            <MusicNote sx={{fontSize: 300, color: 'white'}} />  
          </Grid>
          <Grid item xs={12}>
            <Typography color="white" variant='h1'>Welcome to Music App</Typography>  
        </Grid>
     </Grid>
  )
}


