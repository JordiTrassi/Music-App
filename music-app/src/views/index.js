export * from './AlbumsGridView';
export * from './AlbumsListView';
export * from './InitialView';