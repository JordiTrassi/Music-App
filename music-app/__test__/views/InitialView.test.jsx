import { screen, render } from '@testing-library/react';
import { InitialView } from '../../src/views/InitialView';
import { BrowserRouter as Router } from 'react-router-dom';


describe('Pruebas en <InitialView.jsx>', () => {

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(
            <Router>
                <InitialView />
            </Router>
        );
        
        expect(container).toMatchSnapshot();
        
    });

    test('Debe de mostar el titulo "Welcome to Music App" ', () => { 
        render(
            <Router>
                <InitialView />
            </Router>
        );

        expect(screen.getByText("Welcome to Music App")).toBeTruthy();
    });
    
});