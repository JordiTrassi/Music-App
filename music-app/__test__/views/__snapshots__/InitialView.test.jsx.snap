// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <InitialView.jsx> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container animate__animated animate__fadeIn css-6paxfn-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-pv8kf-MuiGrid-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-7hcj6y-MuiSvgIcon-root"
        data-testid="MusicNoteIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 3v10.55c-.59-.34-1.27-.55-2-.55-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4V7h4V3h-6z"
        />
      </svg>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-pv8kf-MuiGrid-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 css-13gn88f-MuiTypography-root"
      >
        Welcome to Music App
      </h1>
      <h3
        class="MuiTypography-root MuiTypography-h3 css-1qnne57-MuiTypography-root"
      >
        Search your favorite music!
      </h3>
    </div>
  </div>
</div>
`;
